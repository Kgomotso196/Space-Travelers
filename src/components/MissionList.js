import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Mission from './Mission';
import { fetchMissions } from '../redux/missions/missionsSlice';
import './mission.css'

const MissionList = () => {
  const dispatch = useDispatch();
  const missions = useSelector((state) => state.missions.missions);

  useEffect(() => {
    dispatch(fetchMissions());
  }, []);

  console.log(missions);

  return (
    <div>
      <h1>Missions</h1>
      <table>
        <thead>
          <tr>
            <th>Mission</th>
            <th id="description">Description</th>
            <th>Status</th>
            <th></th>
          </tr>
        </thead>
        <tbody>
          {missions.map((mission) => (
            <Mission key={mission.mission_name} mission_name={mission.mission_name} description={mission.description} />
          ))};
        </tbody>
      </table>
    </div>
  )
}

export default MissionList;
