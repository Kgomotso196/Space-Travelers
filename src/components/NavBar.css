:root {
  --background: #050505;
  --text: #fafafa;
  --primary: #1e324d;
  --secondary: #0e1825;
  --accent: #91add4;
}

nav {
  width: 90%;
  margin: 2% auto;
  margin-top: auto;
  margin-bottom: auto;
  padding: 2% 0;
  justify-content: space-between;
  align-items: center;
  display: flex;
  border-bottom: 1px solid var(--accent);
}

nav img {
  width: 85px;
  height: 80px;
  margin-top: -1%;
  margin-right: 2%;
  margin-left: 5%;
  margin-bottom: 1%;
}

nav h1 {
  width: 1500px;
  max-width: 1500px;
  white-space: nowrap;
  overflow: hidden;
  margin-left: 2px;
}

nav ul {
  list-style: none;
  display: flex;
  justify-content: flex-end;
  width: 1500px;
  margin-right: 20px;
}

.nav-item {
  height: min-content;
  margin: 1.5% 2%;
}

.nav-item > * {
  width: max-content;
  text-decoration: none;
  color: var(--text);
}

.nav-item:last-child {
  padding-left: 3%;
  border-left: 1px solid var(--accent);
}

.nav-item *:hover {
  text-decoration: underline;
}

.active {
  text-decoration: underline;
}
