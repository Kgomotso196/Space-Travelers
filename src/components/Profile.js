import React from 'react';

const Profile = () => (
  <div>
    <h1>My Profile</h1>
  </div>
);

export default Profile;
