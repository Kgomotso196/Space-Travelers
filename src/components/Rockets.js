import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchRockets, reserveRocket } from '../redux/rockets/rocketsSlice';
import RocketList from '../components/RocketList';

function Rockets() {
  const dispatch = useDispatch();
  const { rockets, loading } = useSelector((store) => store.rockets);

  useEffect(() => {
    if (rockets.length === 0) {
      dispatch(fetchRockets());
    }
  }, [dispatch, rockets]);

  const clickHandler = (e) => {
    dispatch(reserveRocket(e.target.id));
  };

  if (loading === 'loading') {
    return (
      <h1 style={{ marginLeft: '40px' }}>Loading...</h1>
    );
  }

  return (
    <>
      {rockets.map((item) => (
        <RocketList
          key={item.id}
          id={item.id}
          name={item.name}
          description={item.description}
          flickrImages={item.image}
          reserved={item.reserved}
          onClick={clickHandler}
        />
      ))}
    </>
  );
}

export default Rockets;