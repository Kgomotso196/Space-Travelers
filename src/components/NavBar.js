import React from 'react';
import { NavLink } from 'react-router-dom';
import './NavBar.css';


const NavBar = () =>
  <nav>
    <ul>
      <li className="nav-item"><NavLink to="/" activeclassname="active">Rockets</NavLink></li>
      <li className="nav-item"><NavLink to="/missions" activeclassname="active"><p>Missions</p></NavLink></li>
      <li className="nav-item"><NavLink to="/profile" activeclassname="active"><p>My Profile</p></NavLink></li>
    </ul>
  </nav>;

export default NavBar;
