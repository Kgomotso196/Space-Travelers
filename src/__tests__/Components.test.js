import React from 'react';
import { render } from '@testing-library/react';
import MissionList from '../components/MissionList';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/store';
import Rockets from '../components/Rockets';
import Profile from '../components/Profile';
import NavBar from '../components/NavBar';

describe('Component testing', () => {
  test('Mission list should render correctly', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Router>
          <MissionList />
        </Router>
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('Rocket list should render correctly', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Router>
          <Rockets />
        </Router>
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('Profile should render correctly', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Router>
          <Profile />
        </Router>
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('NavBar should render correctly', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Router>
          <NavBar />
        </Router>
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
