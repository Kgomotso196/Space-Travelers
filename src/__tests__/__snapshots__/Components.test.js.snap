// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component testing Mission list should render correctly 1`] = `
<DocumentFragment>
  <div>
    <table>
      <thead>
        <tr>
          <th>
            Mission
          </th>
          <th
            id="description"
          >
            Description
          </th>
          <th
            id="status"
          >
            Status
          </th>
          <th />
        </tr>
      </thead>
      <tbody />
    </table>
    ;
  </div>
</DocumentFragment>
`;

exports[`Component testing NavBar should render correctly 1`] = `
<DocumentFragment>
  <nav>
    <img
      alt="planet logo"
      src="planet.png"
    />
    <h1>
      Space Travelers' Hub
    </h1>
    <ul>
      <li
        class="nav-item"
      >
        <a
          activeclassname="active"
          class=""
          href="/rockets"
        >
          Rockets
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          activeclassname="active"
          class=""
          href="/missions"
        >
          Missions
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          activeclassname="active"
          class=""
          href="/profile"
        >
          My Profile
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`Component testing Profile should render correctly 1`] = `
<DocumentFragment>
  <div
    id="profile-container"
  >
    <div
      class="list"
    >
      <h2>
        My Missions
      </h2>
      <ul
        id="missions"
      >
        <li>
          You have not joined any missions yet
        </li>
      </ul>
    </div>
    <div
      class="list"
    >
      <h2>
        My Rockets
      </h2>
      <ul
        id="rockets"
      >
        <li>
          You have not reserved any rockets yet
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Component testing Rocket list should render correctly 1`] = `<DocumentFragment />`;
